snuba/subscriptions/data.py

```diff
@@ -74,20 +74,53 @@ def from_string(cls, value: str) -> SubscriptionIdentifier:
         return cls(PartitionId(int(partition)), UUID(uuid))
 
 
-@dataclass(frozen=True)
-class SubscriptionData:
-    """
-    Represents the state of a subscription.
-    """
-
+@dataclass(frozen=True, kw_only=True)
```

Comment:
Adding the `kw_only` arg so we can specify a default for tenant_ids in the base class here

>If true (the default value is False), then all fields will be marked as keyword-only. If a field is marked as keyword-only, then the only affect is that the __init__() parameter generated from a keyword-only field must be specified with a keyword when __init__() is called.